common/src/main/java/xyz/jpenilla/squaremap/common/IconRegistry.java

```diff
@@ -23,9 +22,6 @@ public final class IconRegistry implements Registry<BufferedImage> {
     public IconRegistry(final DirectoryProvider directoryProvider) {
         this.directory = directoryProvider.webDirectory().resolve("images/icon/registered/");
         try {
-            if (Files.exists(this.directory)) {
-                FileUtil.deleteRecursively(this.directory);
```

Comment:
I'd prefer this to be left alone, we only need to remove deletion from `#unregister` to resolve the linked issue.

Reply:
Yes and no. If we unlink files and directories on startup, they will *briefly* become unavailable to web clients and may cause 404s. Generally this window will be very short, but it can be longer. If the admin neglects to update a squaremap plugin, the plugin won't load. The plugin's registered icons will be eliminated without replacement, but its marker data might remain. Granted, this might motivate the admin to investigate the failure.

Are we recursively unlinking this directory just for cleanup purposes or are there other concerns?

Reply:
> If the admin neglects to update a squaremap plugin, the plugin won't load. The plugin's registered icons will be eliminated without replacement, but its marker data might remain. Granted, this might motivate the admin to investigate the failure.

This is the desired behavior, an uninstalled (or equivalent) addon shouldn't have data persist in squaremap. The marker data won't remain, it will be gone by the first run of the marker update task.

So yes, it's for cleanup purposes. I'd really like to avoid having unused files sit around potentially forever (even if they are a few kb in size).

One way to satisfy that and also avoid even a brief window for 404s could be:
- Keep track of all icon keys that were used during the runtime of squaremap (unregistering doesn't remove from this list)
- On shutdown delete files for keys that weren't used during the entire runtime

Reply:
A usage check would be the most correct approach. We'd have to keep a `Set` of filenames for this purpose. It would be simpler to do a non-recursive delete if we can guarantee the `registered/` directory will be flat.

Regardless, that is probably beyond the scope of this PR, so we'll leave the existing startup delete for now.